### PR TITLE
Minor Adjustments and Improvements

### PR DESCRIPTION
**Pull request by brubraz**
### Changes

- Implemented the Fisher-Yates Shuffle algorithm to enhance randomness when shuffling the deck.
- Adjusted button alignment on the "LOOKHAND" pop-up.
- Added @brubraz as a contributor to access and view bug reports.
- Suppress Ambush trigger if no valid target exists.
- Quilan: Suppress ability's trigger when no valid target is available.

### CODE REVIEW

File: CardGetters.php [hunk 1]
```diff
@@ -318,7 +318,8 @@ function &GetAllies($player)
 function &GetTheirAllies($player)
 {
   global $p1Allies, $p2Allies;
-  return $player == 1 ? $p2Allies : $p1Allies;
```
Comment (mariotee) [inline CardGetters.php:321]: what was wrong with this ternary operator?

File: Libraries/NetworkingLibraries.php [hunk 2]
```diff
@@ -1617,7 +1648,7 @@ function GetTargetOfAttack($attackID)
           AddDecisionQueue("SETDQVAR", $mainPlayer, 0, 1);
           AddDecisionQueue("PASSPARAMETER", $mainPlayer, "Units", 1);
         }
-        AddDecisionQueue("SPECIFICCARD", $mainPlayer, "MAUL_TWI,$attackerIndex", 1);
+        AddDecisionQueue("SPECIFICCARD", $mainPlayer, "MAUL_TWI," . $attacker->Index(), 1);
```
Comment (mariotee) [inline Libraries/NetworkingLibraries.php:1651]: was this path tested? i would also look into ECL'ing Maul and using Maul Leader to use Overwhelm with Maul

File: StartEffects.php [hunk 3]
```diff
@@ -37,10 +37,8 @@
 $initiativePlayer = $firstPlayer;
 $intiativeTaken = 0;
 
-for ($i=0; $i<2; $i++) { // Improve randomness
```
[the remark this answers is not in the document]
Comment (brubraz) [reply StartEffects.php:40]: I had set it to shuffle twice to improve randomness, but the result wasn't satisfactory. After some research, I found that it's more efficient to shuffle just once using the Fisher-Yates method, which is more effective in ensuring an unbiased shuffle.

File: CardGetters.php (hunk 1 above)
Comment (brubraz) [reply CardGetters.php:321]: The PHP code was issuing a warning: "Only variable references should be returned by reference." For some reason, when using the ternary operator, it didn't recognize that the return was a variable reference.

Comment (mariotee) [reply CardGetters.php:321]: warning in your editor or in the browser?

File: Libraries/NetworkingLibraries.php (hunk 2 above)
Comment (mariotee) [reply Libraries/NetworkingLibraries.php:1651]: okay i did some tests. Maul looks good. i'm going to also test how this plays with my recent changes for when closing the combat chain state. we're getting issues on ping-first,shoot-first interactions

Comment (brubraz) [reply Libraries/NetworkingLibraries.php:1651]: Yes, I made sure to carefully review all instances of these changes and also tested all scenarios locally.

File: CardGetters.php (hunk 1 above)
Comment (brubraz) [reply CardGetters.php:321]: Warning in the debugger

Comment (mariotee) [reply CardGetters.php:321]: hm.. that's odd. i'm not getting any warning in VSCode

Comment (brubraz) [reply CardGetters.php:321]: Are you running the debug mode on VSCode?

Comment (mariotee) [reply CardGetters.php:321]: yes

Comment (brubraz) [reply CardGetters.php:321]: You can replicate the warning here https://onlinephp.io/

Example:
```
<?php

function &GetTheirAllies($player) {
  $p1Allies = [1, 2, 3];
  $p2Allies = [4, 5, 6];
  return $player == 1 ? $p2Allies : $p1Allies;
}

$var1 = &GetTheirAllies(1);

echo($var1[0]);
```

image

Comment (mariotee) [reply CardGetters.php:321]: ahh okay. interesting..

**Pull request merged**